src/database/queries/requestSessions.js

```diff
@@ -0,0 +1,253 @@
+		throw error
+	}
+}
+
+exports.getpendingRequests = async (userId, page, pageSize) => {
+	try {
+		const currentPage = Number.isInteger(page) && page > 0 ? page : 1
+		const limit = Number.isInteger(pageSize) && pageSize > 0 ? pageSize : 10
+		const offset = (currentPage - 1) * limit
+
+		const result = await requestSession.findAndCountAll({
+			where: {
+				user_id: userId,
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+			},
+			raw: true,
+			limit,
+			offset,
+		})
+
+		return result
+	} catch (error) {
+		throw error
+	}
+}
+
+exports.approveRequest = async (userId, friendId, startDate, endDate, sessionId) => {
+	const t = await sequelize.transaction() // start transaction
+
+	try {
+		const updateData = {
+			status: common.CONNECTIONS_STATUS.ACCEPTED,
+			session_id: sessionId,
+			created_by: friendId,
+			updated_by: userId,
+		}
+
+		const requests = await requestSession.update(updateData, {
+			where: {
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+				created_by: friendId,
+				start_date: startDate,
+				end_date: endDate,
+			},
+			transaction: t,
+			individualHooks: true,
+		})
+
+		await t.commit()
+		return requests
+	} catch (error) {
+		await t.rollback()
+		throw error
+	}
+}
+
+exports.rejectRequest = async (userId, friendId, rejectReason, startDate, endDate) => {
+	try {
+		let updateData = {
+			status: common.CONNECTIONS_STATUS.REJECTED,
+			updated_by: userId,
+		}
+
+		if (rejectReason) {
+			updateData.reject_reason = { reason: rejectReason }
+		}
+
+		return await requestSession.update(updateData, {
+			where: {
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				created_by: friendId,
+				start_date: startDate,
+				end_date: endDate,
+			},
+			individualHooks: true,
+		})
+	} catch (error) {
+		throw error
+	}
+}
+exports.findOneRequest = async (userId, friendId, startDate, endDate) => {
+	try {
+		const sessionRequest = await requestSession.findOne({
+			where: {
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+				created_by: friendId,
+				start_date: startDate,
+				end_date: endDate,
+			},
+			raw: true,
+		})
+
+		return sessionRequest
+	} catch (error) {
+		throw error
+	}
+}
+
+exports.checkPendingRequest = async (userId, friendId) => {
+	try {
+		const result = await requestSession.findAndCountAll({
+			where: {
+				user_id: userId,
+				friend_id: friendId,
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+			},
+		})
+		return result
+	} catch (error) {
+		throw error
+	}
+}
+
+exports.getRequestSessions = async (userId, friendId, startDate, endDate, status) => {
+	try {
+		const whereClause = {
```

Comment:
@sumanvpacewisdom to get requestSession frontEnd has to pass the request Session Id